homeassistant/components/binary_sensor/mystrom.py

```diff
@@ -0,0 +1,98 @@
+"""
+Support for the myStrom buttons.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.mystrom/
+"""
+import asyncio
+import logging
+
+from homeassistant.const import HTTP_UNPROCESSABLE_ENTITY
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.components.binary_sensor import (BinarySensorDevice, DOMAIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['http']
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up myStrom Binary Sensor."""
+    button = None
+
+    hass.http.register_view(MyStromView(button, async_add_devices))
+
+    return True
+
+
+class MyStromView(HomeAssistantView):
+    """View to handle requests from myStrom buttons."""
+
+    url = '/api/mystrom'
+    name = 'api:mystrom'
+
+    def __init__(self, button, add_devices):
+        """Initialize the myStrom URL endpoint."""
+        self.button = button
```

Comment:
We can't just have 1 button per view, as many different myStrom buttons can hit this view. Make it a list.